libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -0,0 +1,614 @@
+        break;
+    }
+
+    case SiyiCommandId::FUNCTION_FEEDBACK_INFO: {
+        if (_parsed_msg.data_bytes_received != 1) {
+            unexpected_len = true;
+            break;
+        }
+        const uint8_t fund_feedback_info = _msg_buff[_msg_buff_data_start];
+        const char* err_prefix = "Mount: Siyi";
+        switch (fund_feedback_info) {
+        case 0:
+            debug("FnFeedB success");
+            break;
+        case 1:
+            gcs().send_text(MAV_SEVERITY_ERROR, "%s failed to take picture", err_prefix);
+            break;
+        case 2:
+            debug("HDR on");
+            break;
+        case 3:
+            debug("HDR off");
+            break;
+        case 4:
+            gcs().send_text(MAV_SEVERITY_ERROR, "%s failed to record video", err_prefix);
+            break;
+        default:
+            debug("FnFeedB unexpected val:%u", (unsigned)fund_feedback_info);
+        }
+        break;
+    }
+
+    case SiyiCommandId::PHOTO:
+        // no ack should ever be sent by the gimbal
+        break;
+
+    case SiyiCommandId::ACQUIRE_GIMBAL_ATTITUDE: {
+        if (_parsed_msg.data_bytes_received != 12) {
+            unexpected_len = true;
+            break;
+        }
+        _last_current_angle_rad_ms = AP_HAL::millis();
+        _current_angle_rad.z = -radians((int16_t)UINT16_VALUE(_msg_buff[_msg_buff_data_start+1], _msg_buff[_msg_buff_data_start]) * 0.1);   // yaw angle
+        _current_angle_rad.y = radians((int16_t)UINT16_VALUE(_msg_buff[_msg_buff_data_start+3], _msg_buff[_msg_buff_data_start+2]) * 0.1);  // pitch angle
+        _current_angle_rad.x = radians((int16_t)UINT16_VALUE(_msg_buff[_msg_buff_data_start+5], _msg_buff[_msg_buff_data_start+4]) * 0.1);  // roll angle
+        //const float yaw_rate_degs = -(int16_t)UINT16_VALUE(_msg_buff[_msg_buff_data_start+7], _msg_buff[_msg_buff_data_start+6]) * 0.1;   // yaw rate
+        //const float pitch_rate_deg = (int16_t)UINT16_VALUE(_msg_buff[_msg_buff_data_start+9], _msg_buff[_msg_buff_data_start+8]) * 0.1;   // pitch rate
+        //const float roll_rate_deg = (int16_t)UINT16_VALUE(_msg_buff[_msg_buff_data_start+11], _msg_buff[_msg_buff_data_start+10]) * 0.1;  // roll rate
+        break;
+    }
+
+    default:
+        debug("Unhandled CmdId:%u", (unsigned)_parsed_msg.command_id);
+        break;
+    }
+
+    // handle unexpected data buffer lenth
+    if (unexpected_len) {
+        debug("CmdId:%u unexpected len:%u", (unsigned)_parsed_msg.command_id, (unsigned)_parsed_msg.data_bytes_received);
+    }
+}
+
+// methods to send commands to gimbal
+void AP_Mount_Siyi::send_packet(SiyiCommandId cmd_id, const uint8_t* databuff, uint8_t databuff_len)
+{
+    // calculate packet size
+    const uint8_t packet_size = AP_MOUNT_SIYI_PACKETLEN_MIN + databuff_len;
+
+    // check for sufficient space in outgoing buffer
+    if (_uart->txspace() < packet_size) {
+        return;
```

Comment:
```suggestion
        return false;
```

we can report an error, so should....

Reply:
ok, send_packet returns a boolean which will be fed back to the user if triggered via an RC switch.